/**
 * Copyright (c) 2015 - The CM Authors <legal@clickmatcher.com>
 *   All Rights Reserved.
 *
 * This file is CONFIDENTIAL -- Distribution or duplication of this material or
 * the information contained herein is strictly forbidden unless prior written
 * permission is obtained.
 */
package csql;

message RemoteAggregateParams {
  message SelectListItem {
    required string expression = 1;
    optional string alias = 2;
  };

  required string table_name = 1;
  repeated SelectListItem aggregate_expression_list = 2;
  repeated SelectListItem select_expression_list = 3;
  repeated string group_expression_list = 4;
  optional string where_expression = 5;
  required uint64 aggregation_strategy = 6;
};
